lib/gosu/compat.rb

```diff
@@ -101,6 +101,70 @@ def self.from_text(*args)
       from_text_without_window(*args)
     end
   end
+
+  alias _draw_rot draw_rot
+  def draw_rot(*args)
+    Gosu.deprecation_message("Image#draw_rot is deprecated in Gosu x.y, use draw with an options hash instead, see https://www.libgosu.org/rdoc/Gosu/Image.html.")
+    _draw_rot(*args)
+  end
+
+  alias _draw draw
+  DRAW_ALIGNMENTS = {
+    :upper_left => [-1, -1],
+    :up => [0, -1],
+    :upper_right => [1, -1],
+    :left => [-1, 0],
+    :middle => [0, 0],
+    :right => [1, 0.0],
+    :bottom_left => [-1, 1],
+    :bottom => [0, 1],
+    :bottom_right => [1, 1],
+  }
+  def draw(x, y, z, *args)
```

Comment:
I wonder if executing relatively much Ruby code in every `Image#draw` call slows down existing Gosu games. I guess we should release a 0.14.0-pre1 gem this time and rewrite `Image#draw` in C++ if this is really an issue

Reply:
Ok my girls are sleeping againg.....So - you're right. Performance might really be an issue here as drawing is done for possibly thousands of things. I use Ruby for over 10 years now and it was always fast enough with pure Ruby code or existing gems but I tend to forget that *this* is not my game but the layer below which should be fast. I'll see if I can imitate what you already did with the other methods supporting options.